lib/ansible/module_utils/ecs/api.py

```diff
@@ -149,45 +150,34 @@ def restmethod(self, *args, **kwargs):
             # modify the URL to add path parameters
             url = url + "?" + urlencode(query_parameters)
 
-        # Try up to 3 times, if receiving a 429 'too many requests' error code
-        for i in range(1, 3):
+        try:
             if body_parameters:
```

Comment:
All other changes were related to syntax and code review comments, but I did make changes to this block of code to 

1. properly handle the errors thrown by urllib/urllib2
2. Skip the "try multiple times" dance. Many ECS operations, including requesting an SSL certificate are ones you wouldn't *want* to try again if you received an arbitrary exception - that's the point where you would want a playbook to fail so it can be triaged, because it's worse to get multiple certificates without realizing it than to not get a certificate.